Fix typos in manpage revealed by lintian
--- a/src/man/man1/owshell.man
+++ b/src/man/man1/owshell.man
@@ -223,7 +223,7 @@
 .I owserver
 process can be local or remote.
 .PP
-If the server option is not specified, the default is the local machine and the IANA allocated default port of 4304. Thus "-s localhost:4304" is the equivalent.
+If the server option is not specified, the default is the local machine and the IANA allocated default port of 4304. Thus "\-s localhost:4304" is the equivalent.
 .SH DATA OPTIONS
 .SH \-\-hex
 Hexidecimal mode. For reading data, each byte of character will be displayed as two characrters 0-9ABCDEF. Most useful for reading memory locations. No spaces between data.
--- a/module/ownet/perl5/OWNet/lib/OWNet.pm
+++ b/module/ownet/perl5/OWNet/lib/OWNet.pm
@@ -451,8 +451,6 @@
 
 =item 2 No B<owserver> at I<address>
 
-=item
-
 =back
 
 =cut
@@ -502,8 +500,6 @@
 
 =item 4 I<path> not a readable device property
 
-=item
-
 =back
 
 =cut
@@ -554,8 +550,6 @@
 
 =item 5 I<value> incorrect size or format
 
-=item
-
 =back
 
 =cut
@@ -605,8 +599,6 @@
 
 =item 4 I<path> not a directory
 
-=item
-
 =back
 
 =cut
@@ -682,8 +674,6 @@
 
 =item 4 I<path> not a device
 
-=item
-
 =back
 
 =cut
