Do not use -rpath for /usr/lib
--- a/module/swig/perl5/OW/Makefile.linux.in
+++ b/module/swig/perl5/OW/Makefile.linux.in
@@ -1,5 +1,6 @@
 #!/usr/bin/perl -w
 
+use Config ;
 use ExtUtils::MakeMaker ;
 
 WriteMakefile(
@@ -12,8 +13,10 @@
     'INC'       =>	q[-I../../../../src/include -I../../../owlib/src/include @CPPFLAGS@],
 # Default value for LDDLFLAGS is $Config{lddlflags}="-shared -L/usr/local/lib"
 # but we want rpath to be @libdir@ or @exec_prefix@/lib
-    'LDDLFLAGS' =>      q[-shared -L../../../owlib/src/c/.libs -Wl,-rpath=@prefix@/lib],
-    'LIBS'      =>	q[-L../../../owlib/src/c/.libs -low @PTHREAD_LIBS@ @LIBS@],
+# Also adding flags for hardening (Debian)
+    'LDDLFLAGS' =>      "$ENV{DEB_LDFLAGS} -shared -L../../../owlib/src/c/.libs",
+    'CCFLAGS'   =>      "$Config{ccflags} $ENV{DEB_CFLAGS} $ENV{DEB_CPPFLAGS}",
+    'LIBS'      =>	q[-L../../../owlib/src/c/.libs -low],
     'OBJECT'	=>	'ow_wrap.o',
     'NAME'      =>	'OW',
 	'POLLUTE'	=>  1,
--- a/module/swig/perl5/Makefile.am
+++ b/module/swig/perl5/Makefile.am
@@ -18,6 +18,8 @@
 else
 if HAVE_DEBIAN
 	( cd OW; $(PERL) Makefile.PL INSTALLDIRS=vendor )
+	cd OW ; for i in `grep -wl ^LD_RUN_PATH Makefile Makefile.[^P]*` ; do sed -i 's@^LD_RUN_PATH.*@LD_RUN_PATH=@' $$i ; done
+
 else
 	( cd OW; $(PERL) Makefile.PL )
 if HAVE_FREEBSD
--- a/module/owcapi/src/c/Makefile.am
+++ b/module/owcapi/src/c/Makefile.am
@@ -11,7 +11,7 @@
 # Debian patch
  libowcapi_la_LDFLAGS = -low -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} ${LD_EXTRALIBS} -shared -shrext .so
 else
- libowcapi_la_LDFLAGS = -low -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} ${LIBUSB_LIBS} ${LD_EXTRALIBS} -shared -shrext .so
+ libowcapi_la_LDFLAGS = -low -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} ${LD_EXTRALIBS} -shared -shrext .so
 endif
 endif
 
--- a/module/owserver/src/c/Makefile.am
+++ b/module/owserver/src/c/Makefile.am
@@ -42,11 +42,9 @@
 	-Wcast-align \
 	-Wstrict-prototypes \
 	-Wredundant-decls \
-	${EXTRACFLAGS} \
-	${PTHREAD_CFLAGS} \
-	${LIBUSB_CFLAGS}
+	${EXTRACFLAGS}
 
-LDADD = -low ${LIBUSB_LIBS} ${PTHREAD_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
+LDADD = -low ${LD_EXTRALIBS} ${OSLIBS}
 
 #if HAVE_CYGWIN
 #NOWINE=1
--- a/module/owftpd/src/c/Makefile.am
+++ b/module/owftpd/src/c/Makefile.am
@@ -23,9 +23,7 @@
 	-Wcast-align \
 	-Wstrict-prototypes \
 	-Wredundant-decls \
-	${EXTRACFLAGS} \
-	${LIBUSB_CFLAGS} \
-	${PTHREAD_CFLAGS}
+	${EXTRACFLAGS}
 
-LDADD = -low ${LIBUSB_LIBS} ${PTHREAD_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
+LDADD = -low ${LD_EXTRALIBS} ${OSLIBS}
 
--- a/module/owhttpd/src/c/Makefile.am
+++ b/module/owhttpd/src/c/Makefile.am
@@ -23,9 +23,7 @@
 	-Wcast-align \
 	-Wstrict-prototypes \
 	-Wredundant-decls \
-	${EXTRACFLAGS} \
-	${PTHREAD_CFLAGS} \
-	${LIBUSB_CFLAGS}
+	${EXTRACFLAGS}
 
-LDADD = -low ${LIBUSB_LIBS} ${PTHREAD_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
+LDADD = -low ${LD_EXTRALIBS} ${OSLIBS}
 
--- a/module/owshell/src/c/Makefile.am
+++ b/module/owshell/src/c/Makefile.am
@@ -43,5 +43,5 @@
 	-Wredundant-decls \
 	${EXTRACFLAGS}
 
-LDADD = ${LD_EXTRALIBS} ${OSLIBS}
+LDADD = ${DL_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
 
--- a/module/owlib/src/c/Makefile.am
+++ b/module/owlib/src/c/Makefile.am
@@ -218,7 +218,7 @@
 if HAVE_CYGWIN
 libow_la_LDFLAGS = ${PTHREAD_LIBS} -shared -no-undefined ${LIBUSB_LIBS} ${LD_EXTRALIBS}
 else
-libow_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} -shared -shrext .so ${LIBUSB_LIBS} ${LD_EXTRALIBS}
+libow_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} -shared -shrext .so ${LIBUSB_LIBS} ${LD_EXTRALIBS} ${M_LIBS} ${DL_LIBS}
 endif
 
 # Maybe need this for MacOS X
--- a/module/owfs/src/c/Makefile.am
+++ b/module/owfs/src/c/Makefile.am
@@ -18,8 +18,7 @@
 	-Wstrict-prototypes \
 	-Wredundant-decls \
 	${EXTRACFLAGS} \
-	${LIBUSB_CFLAGS} \
 	${PTHREAD_CFLAGS}
 
-LDADD = -low ${FUSE_LIBS} ${LIBUSB_LIBS} ${PTHREAD_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
+LDADD = -low ${FUSE_LIBS} ${PTHREAD_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
 
--- a/module/ownet/c/src/c/Makefile.am
+++ b/module/ownet/c/src/c/Makefile.am
@@ -30,7 +30,7 @@
 if HAVE_CYGWIN
 libownet_la_LDFLAGS = ${PTHREAD_LIBS} ${LD_EXTRALIBS} -shared -no-undefined
 else
-libownet_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} ${LD_EXTRALIBS} -shared -shrext .so
+libownet_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) -release $(LT_RELEASE) ${PTHREAD_LIBS} ${LD_EXTRALIBS} -shared -shrext .so ${DL_LIBS}
 endif
 
 # Maybe need this for MacOS X
--- a/module/swig/python/Makefile.am
+++ b/module/swig/python/Makefile.am
@@ -8,9 +8,8 @@
 ow_wrap.c: ../ow.i ${LIBOW}
 	$(SWIG) -python -o $@ ../ow.i
 
-OW.py: ow_wrap.c setup.py
-	LDFLAGS="$(PYLDFLAGS) ${PTHREAD_LIBS} ${LIBUSB_LIBS} ${LD_EXTRALIBS}" CFLAGS="$(PYCFLAGS) $(PIC_FLAGS)" $(PYTHON) setup.py build
-#	$(PYTHON) setup.py build
+OW.py: ow_wrap.c setup.py $(LIBOW)
+	CFLAGS="" $(PYTHON) setup.py build
 
 install-data-local:
 #	OpenSUSE is buggy and install libraries at /usr/local.
--- a/configure.ac
+++ b/configure.ac
@@ -726,9 +726,6 @@
 	m4_include(module/swig/python/python.m4)
 	AC_MSG_RESULT(Looking for location of Python executable)
 	SC_PATH_PYTHON
-	AC_SUBST(PYCFLAGS)
-	AC_SUBST(PYLDFLAGS)
-	AC_SUBST(PYLIB)
 	AC_SUBST(PYVERSION)
 	AC_SUBST(PYSITEDIR)
 	AC_SUBST(PYTHONDYNAMICLINKING)
@@ -1569,11 +1566,23 @@
 AC_TYPE_SIGNAL
 AC_CHECK_FUNCS([accept daemon getaddrinfo freeaddrinfo gethostbyname2_r gethostbyaddr_r gethostbyname_r getservbyname_r getopt getopt_long gettimeofday inet_ntop inet_pton memchr memset select socket strcasecmp strchr strdup strncasecmp strtol strtoul twalk tsearch tfind tdelete tdestroy vasprintf strsep vsprintf vsnprintf writev getline])
 
+save_LIBS="$LIBS"
+DL_LIBS=""
 if test "${ENABLE_ZERO}" = "true" ; then
-	AC_SEARCH_LIBS(dlopen, dl, AC_DEFINE(HAVE_DLOPEN, 1, [Define if you have dlopen]))
+	AC_SEARCH_LIBS(dlopen, dl,
+		DL_LIBS="-ldl"
+		AC_DEFINE(HAVE_DLOPEN, 1, [Define if you have dlopen]))
 fi
+AC_SUBST([DL_LIBS])
+LIBS="$save_LIBS"
 
-AC_SEARCH_LIBS(lrint, m, AC_DEFINE(HAVE_LRINT, 1, [Define if you have lrint]))
+save_LIBS="$LIBS"
+M_LIBS=""
+AC_SEARCH_LIBS(lrint, m,
+       M_LIBS="-lm"
+       AC_DEFINE(HAVE_LRINT, 1, [Define if you have lrint]))
+AC_SUBST([M_LIBS])
+LIBS="$save_LIBS"
 
 AC_SEARCH_LIBS(nanosleep, rt posix4, AC_DEFINE(HAVE_NANOSLEEP, 1, [Define if you have nanosleep]))
 # clock_gettime needed if real-time wait is needed for sem_timedwait. gettimeofday is enough.
--- a/module/swig/php/Makefile.am
+++ b/module/swig/php/Makefile.am
@@ -25,7 +25,7 @@
 	${PTHREAD_CFLAGS} \
 	@PHPINC@
 
-LDADD = -L../../owlib/src/c -low ${PTHREAD_LIBS} ${LD_EXTRALIBS} ${OSLIBS}
+LDADD = -L../../owlib/src/c -low
 
 nodist_libowphp_la_SOURCES = ow_wrap.c php_OW.h
 
--- a/module/swig/python/setup.py.in
+++ b/module/swig/python/setup.py.in
@@ -73,9 +73,6 @@
     print >> out, "extra_objects", extra_objects
     out.close()
 
-
-enable_usb = '@ENABLE_USB@'
-enable_mt = '@ENABLE_MT@'
 have_darwin = '@HAVE_DARWIN@'
 have_freebsd = '@HAVE_FREEBSD@'
 
@@ -89,42 +86,12 @@
 # supporting MacOSX.
 
 my_extra_compile_args = [ '-D_FILE_OFFSET_BITS=64' ]
-if len('@PYCFLAGS@') > 0:
-	my_extra_compile_args = my_extra_compile_args + string.split('@PYCFLAGS@', ' ')
-
-if len('@PIC_FLAGS@') > 0:
-	my_extra_compile_args = my_extra_compile_args + string.split('@PIC_FLAGS@', ' ')
 
 my_extra_link_args = [ ]
-if enable_usb == 'true':
-    if len('@LIBUSB_CFLAGS@') > 1:
-	my_extra_compile_args = my_extra_compile_args + string.split('@LIBUSB_CFLAGS@', ' ')
-
-if enable_usb == 'true':
-    if len('@LIBUSB_LIBS@') > 1:
-	my_extra_link_args = my_extra_link_args + string.split('@LIBUSB_LIBS@', ' ')
-
-if enable_mt == 'true':
-    if len('@PTHREAD_CFLAGS@') > 1:
-	my_extra_compile_args = my_extra_compile_args + string.split('@PTHREAD_CFLAGS@', ' ')
-
-if enable_mt == 'true':
-    if len('@PTHREAD_LIBS@') > 1:
-	my_extra_link_args = my_extra_link_args + string.split('@PTHREAD_LIBS@', ' ')
 
 if bsd_fixes:
     my_extra_link_args = my_extra_link_args + string.split('../../owlib/src/c/.libs/libow.so', ' ')
 
-if len('@PYLDFLAGS@') > 1:
-	my_extra_link_args = my_extra_link_args + string.split('@PYLDFLAGS@', ' ')
-
-ARCH=""
-my_platform = '@LIBPOSTFIX@'
-#my_platform = [ platform.machine() ]
-if my_platform == "x86_64" or my_platform == "amd64" or my_platform == "64":
-    print "64bit system detected"
-    ARCH = "x64"
-
 my_libraries = [ 'ow' ]
 
 #if ARCH == "x64":
@@ -164,7 +131,7 @@
 my_extra_link_args = [ arg for arg in my_extra_link_args if len(arg) > 1 ]
 my_include_dirs = [ '../../owlib/src/include', '../../../src/include' ]
 my_library_dirs = [ '../../owlib/src/c/.libs' ]
-my_runtime_library_dirs = [ '@LIBDIR@' ]
+my_runtime_library_dirs = [ ]
 
 sources = [ 'ow_wrap.c' ]
 
